File: String/Paldromic Substring/Python/Brute Force.py
# Simple but inefficient for large strings
def longest_palindrome_brute_force(s):
    # Finds the longest palindromic substring using brute force.
    if not s or len(s) <= 1:  # Handle empty or single-character strings
        return s

    longest = ""  # Initialize the longest palindrome

    for i in range(len(s)):  # Iterate through all possible start indices
        for j in range(i, len(s)):  # Iterate through all possible end indices
            sub = s[i:j + 1]  # Extract the substring
            if is_palindrome(sub) and len(sub) > len(longest):  # Check if it's a palindrome and longer than current longest
                longest = sub  # Update longest palindrome
    return longest

def is_palindrome(s):
    # Checks if a string is a palindrome.
    return s == s[::-1]  # Efficiently reverse and compare string

File: String/Paldromic Substring/JS/Brute Force.js
// Brute Force (Simple but inefficient for large strings)
function longestPalindromeBruteForce(s) {
  // Handle edge cases: empty or single-character strings
    if (!s || s.length <= 1) return s;

  let longest = "", longer=0; // Initialize the longest palindrome

  // Iterate through all possible starting positions
    for (let i = 0; i < s.length; i++) {
        // Iterate through all possible ending positions
        for (let j = i; j < s.length; j++) {
            // Extract the substring
            const sub = s.substring(i, j + 1);
            // Check if the substring is a palindrome and longer than the current longest
            if (isPalindrome(sub) && sub.length > longer) {
                longest = sub; // Update the longest palindrome
                longer = longest.length
            }
        }
    }

  return longest; // Return the longest palindrome found
}

// Helper function to check if a string is a palindrome
function isPalindrome(str) {
    // Reverse the string and compare it to the original
    return str === str.split("").reverse().join("");
}